### DISCUSSION TITLE
TypeError on Home Assistant add on 2023.12.3

### DISCUSSION THREAD

**Discussion by Opiekelly** [theengs/gateway#188] [General]
Hello I am running the add on for Home Assistant 2023.12.3.  The docker container will not start and I get this error.
Core
2023.12.3
Supervisor
2023.11.6
Operating System
11.2
Frontend
20231208.2

`➜  / docker logs 39dc63a1cdff
[09:16:26] INFO: Creating TheengsGateway configuration...
[09:16:29] INFO: IDENTITIES:
[09:16:29] INFO: BINDKEYS:
[09:16:29] INFO: Starting TheengsGateway...
Traceback (most recent call last):
  File "/usr/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/usr/local/lib/python3.9/dist-packages/TheengsGateway/__main__.py", line 5, in <module>
    main()
  File "/usr/local/lib/python3.9/dist-packages/TheengsGateway/__init__.py", line 57, in main
    run(configuration, config_path)
  File "/usr/local/lib/python3.9/dist-packages/TheengsGateway/ble_gateway.py", line 591, in run
    asyncio.run(diagnostics(config_path))
  File "/usr/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/usr/local/lib/python3.9/dist-packages/TheengsGateway/diagnose.py", line 172, in diagnostics
    await _adapters()
  File "/usr/local/lib/python3.9/dist-packages/TheengsGateway/diagnose.py", line 155, in _adapters
    _section(adapter, properties)  # type: ignore[arg-type]
  File "/usr/local/lib/python3.9/dist-packages/TheengsGateway/diagnose.py", line 55, in _section
    print(f"| {name:{max_name}} | {value:{max_value}} |")
TypeError: unsupported format string passed to NoneType.__format__
# Theengs Gateway Diagnostics

## Package Versions

| Name               | Value  |
|--------------------|--------|
| Theengs Gateway    | 1.2.0  |
| Theengs Decoder    | 1.6.4  |
| Bleak              | 0.21.1 |
| Bluetooth Clocks   | 0.2.0  |
| Bluetooth Numbers  | 1.1.1  |
| Paho MQTT          | 1.6.1  |
| Bluetooth Adapters | 0.16.2 |

## Python

| Name           | Value               |
|----------------|---------------------|
| Version        | 3.9.2               |
| Implementation | CPython             |
| Compiler       | GCC 10.2.1 20210110 |
| Executable     | /usr/bin/python3    |

## Operating System

| Name         | Value                                       |
|--------------|---------------------------------------------|
| System       | Linux                                       |
| Release      | 6.1.58-haos-raspi                           |
| Version      | #1 SMP PREEMPT Mon Dec  4 15:50:50 UTC 2023 |
| Machine type | aarch64                                     |

## Configuration

File: /root/theengsgw.conf

```
{
    "adapter": "",
    "bindkeys": {},
    "ble_scan_time": 10,
    "ble_time_between_scans": 60,
    "discovery": "0",
    "discovery_device_name": "TheengsGateway",
    "discovery_filter": "foo",
    "discovery_topic": "homeassistant",
    "enable_tls": 0,
    "enable_websocket": 0,
    "hass_discovery": 1,
    "host": "192.168.0.128",
    "identities": {},
    "log_level": "DEBUG",
    "lwt_topic": "home/TheengsGateway/LWT",
    "pass": "***",
    "port": 1883,
    "presence": 0,
    "presence_topic": "home/presence/TheengsGateway",
    "publish_advdata": 0,
    "publish_all": 1,
    "publish_topic": "home/TheengsGateway/BTtoMQTT",
    "scanning_mode": "active",
    "subscribe_topic": "home/+/BTtoMQTT/undecoded",
    "time_format": "1",
    "time_sync": [],
    "user": "***"
}
```

## Bluetooth adapters

Default adapter: hci0

### hci0

| Name         | Value               |
|--------------|---------------------|
| address      | D8:3A:DD:XX:XX:XX   |
| sw_version   | ha2                 |
| hw_version   | usb:v1D6Bp0246d0546 |
| passive_scan |                   1 |
➜  /`

**Comment by 1technophile** [comment]
Hello,

I think `bindkeys` and `identities` should be with `""` instead of `{}`
Here is an extract of my configuration
![image](https://github.com/theengs/gateway/assets/12672732/99fab526-bee9-4820-a615-791e962f43d5)

**Reply by Opiekelly** [reply]
Well I made the change in the HA UI, however it still fails.

`➜  ~ docker logs 8c6b8bc9327a
[14:31:23] INFO: Creating TheengsGateway configuration...
[14:31:26] INFO: IDENTITIES: " "
[14:31:26] INFO: BINDKEYS: " "
[14:31:26] INFO: Starting TheengsGateway...
Traceback (most recent call last):
  File "/usr/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/usr/local/lib/python3.9/dist-packages/TheengsGateway/__main__.py", line 5, in <module>
    main()
  File "/usr/local/lib/python3.9/dist-packages/TheengsGateway/__init__.py", line 57, in main
    run(configuration, config_path)
  File "/usr/local/lib/python3.9/dist-packages/TheengsGateway/ble_gateway.py", line 591, in run
    asyncio.run(diagnostics(config_path))
  File "/usr/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/usr/local/lib/python3.9/dist-packages/TheengsGateway/diagnose.py", line 172, in diagnostics
    await _adapters()
  File "/usr/local/lib/python3.9/dist-packages/TheengsGateway/diagnose.py", line 155, in _adapters
    _section(adapter, properties)  # type: ignore[arg-type]
  File "/usr/local/lib/python3.9/dist-packages/TheengsGateway/diagnose.py", line 55, in _section
    print(f"| {name:{max_name}} | {value:{max_value}} |")
TypeError: unsupported format string passed to NoneType.__format__
# Theengs Gateway Diagnostics

## Package Versions

| Name               | Value  |
|--------------------|--------|
| Theengs Gateway    | 1.2.0  |
| Theengs Decoder    | 1.6.4  |
| Bleak              | 0.21.1 |
| Bluetooth Clocks   | 0.2.0  |
| Bluetooth Numbers  | 1.1.1  |
| Paho MQTT          | 1.6.1  |
| Bluetooth Adapters | 0.16.2 |

## Python

| Name           | Value               |
|----------------|---------------------|
| Version        | 3.9.2               |
| Implementation | CPython             |
| Compiler       | GCC 10.2.1 20210110 |
| Executable     | /usr/bin/python3    |

## Operating System

| Name         | Value                                       |
|--------------|---------------------------------------------|
| System       | Linux                                       |
| Release      | 6.1.58-haos-raspi                           |
| Version      | #1 SMP PREEMPT Mon Dec  4 15:50:50 UTC 2023 |
| Machine type | aarch64                                     |

## Configuration

File: /root/theengsgw.conf

```
{
    "adapter": "hci0",
    "bindkeys": {
        "INVALID ADDRESS": "***"
    },
    "ble_scan_time": 10,
    "ble_time_between_scans": 60,
    "discovery": "0",
    "discovery_device_name": "TheengsGateway",
    "discovery_filter": "foo",
    "discovery_topic": "homeassistant",
    "enable_tls": 0,
    "enable_websocket": 0,
    "hass_discovery": 1,
    "host": "192.168.0.128",
    "identities": {
        "INVALID ADDRESS": "***"
    },
    "log_level": "DEBUG",
    "lwt_topic": "home/TheengsGateway/LWT",
    "pass": "***",
    "port": 1883,
    "presence": 0,
    "presence_topic": "home/presence/TheengsGateway",
    "publish_advdata": 1,
    "publish_all": 1,
    "publish_topic": "home/TheengsGateway/BTtoMQTT",
    "scanning_mode": "active",
    "subscribe_topic": "home/+/BTtoMQTT/undecoded",
    "time_format": "1",
    "time_sync": [
        "INVALID ADDRESS",
        "INVALID ADDRESS"
    ],
    "user": "***"
}
```

## Bluetooth adapters

Default adapter: hci0

### hci0

| Name         | Value               |
|--------------|---------------------|
| address      | D8:3A:DD:XX:XX:XX   |
| sw_version   | ha2                 |
| hw_version   | usb:v1D6Bp0246d0546 |
| passive_scan |                   1 |
➜  ~`

**Reply by 1technophile** [reply]
Try without a space between the quotes

**Reply by Opiekelly** [reply]
Same problem and the UI seems to remove the "".  Not sure where I can manually configure.

`
```
{
    "adapter": "hci0",
    "bindkeys": {},
    "ble_scan_time": 10,
    "ble_time_between_scans": 60,
    "discovery": "0",
    "discovery_device_name": "TheengsGateway",
    "discovery_filter": "foo",
    "discovery_topic": "homeassistant",
    "enable_tls": 0,
    "enable_websocket": 0,
    "hass_discovery": 1,
    "host": "192.168.0.128",
    "identities": {},
    "log_level": "DEBUG",
    "lwt_topic": "home/TheengsGateway/LWT",
    "pass": "***",
    "port": 1883,
    "presence": 0,
    "presence_topic": "home/presence/TheengsGateway",
    "publish_advdata": 1,
    "publish_all": 1,
    "publish_topic": "home/TheengsGateway/BTtoMQTT",
    "scanning_mode": "active",
    "subscribe_topic": "home/+/BTtoMQTT/undecoded",
    "time_format": "1",
    "time_sync": [],
    "user": "***"
}
'''
`

**Comment by 1technophile** [comment]
Could you try to uninstall and reinstall the addon ?

**Reply by Opiekelly** [reply]
Ok, that resolved the issue.  Thanks for the help.

**Reply by 1technophile** [reply]
Great to hear!
